diff --git a/node_modules/nostr-tools/node_modules/@noble/hashes/utils.js b/node_modules/nostr-tools/node_modules/@noble/hashes/utils.js
index 2ccb60e..b2c57bd 100644
--- a/node_modules/nostr-tools/node_modules/@noble/hashes/utils.js
+++ b/node_modules/nostr-tools/node_modules/@noble/hashes/utils.js
@@ -182,14 +182,11 @@ exports.wrapConstructorWithOpts = wrapConstructorWithOpts;
  * Secure PRNG
  */
 function randomBytes(bytesLength = 32) {
-    if (crypto_1.crypto.web) {
-        return crypto_1.crypto.web.getRandomValues(new Uint8Array(bytesLength));
-    }
-    else if (crypto_1.crypto.node) {
-        return new Uint8Array(crypto_1.crypto.node.randomBytes(bytesLength).buffer);
-    }
-    else {
-        throw new Error("The environment doesn't have randomBytes function");
+    try {
+        const randomBytes = require('expo-random').getRandomBytes;
+        return randomBytes(bytesLength);
+    } catch (e) {
+        throw new Error("The environment doesn't have randomBytes function and we couldn't use expo-random");
     }
 }
 exports.randomBytes = randomBytes;
